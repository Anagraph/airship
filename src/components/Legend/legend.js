import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { shadows, colors } from '../../constants';

/*
  <Legend>
    <Legend.Panel>
      ...
    </Legend.Panel>
  </Legend>
*/

const StyledLegend = styled.div`
  border-radius: 4px;
  background-color: ${colors.ui01};
  width: ${(props) => (props.small ? 160 : 260)}px;
  box-shadow: ${shadows.shadow8};
`;

const StyledPanel = styled.div`
  padding: ${(props) => (props.small ? `18px 10px` : `26px 20px`)};
  position: relative;

  &:before {
    content: '';
    height: 2px;
    left: 0;
    right: 0;
    top: 0;
    background-color: ${colors.ui02};
    position: absolute;
  }

  &:first-child:before {
    display: none;
  }
`;

const Legend = ({ children, small, style }) => {
  return (
    <StyledLegend small={small} style={style}>
      {children}
    </StyledLegend>
  );
};

Legend.Panel = StyledPanel;

Legend.defaultProps = {
  small: false
};

Legend.propTypes = {
  small: PropTypes.bool
};

export default Legend;
