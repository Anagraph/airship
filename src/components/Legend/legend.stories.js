import React from 'react';
import { storiesOf } from '@storybook/react';
import { ThemeProvider } from 'styled-components';
import Legend from './legend';
import Collapsible from '../Collapsible/collapsible';
import Toggle from '../Toggle/toggle';
import Text from '../Typography/text';
import Subheader from '../Typography/subheader';
import { theme } from '../../constants';

const CUSTOM_THEME = {
  ...theme,
  ui01: '#444',
  ui02: '#333',
  type01: 'white',
};

storiesOf('Legend', module)
  .add('Default', () => (
    <div>
      <Legend>
        <Legend.Panel>
          <Collapsible>
            <Collapsible.Header>
              <Subheader>Global Index</Subheader>
            </Collapsible.Header>
            <Collapsible.Content>
              <Toggle htmlFor="active">
                <Text>Active</Text>
              </Toggle>
            </Collapsible.Content>
          </Collapsible>
        </Legend.Panel>
        <Legend.Panel>
          <Collapsible>
            <Collapsible.Header>
              <Subheader>Pattern</Subheader>
            </Collapsible.Header>
            <Collapsible.Content>
              <img alt="" src="http://subtlepatterns2015.subtlepatterns.netdna-cdn.com/patterns/swirl_pattern.png" />
            </Collapsible.Content>
          </Collapsible>
        </Legend.Panel>
      </Legend>
    </div>
  )).add('With custom theme', () => (
    <div>
      <ThemeProvider theme={CUSTOM_THEME}>
        <Legend>
          <Legend.Panel>
            <Collapsible>
              <Collapsible.Header>
                <Subheader>Global Index</Subheader>
              </Collapsible.Header>
              <Collapsible.Content>
                <Toggle htmlFor="active">
                  <Text>Active</Text>
                </Toggle>
              </Collapsible.Content>
            </Collapsible>
          </Legend.Panel>
          <Legend.Panel>
            <Collapsible>
              <Collapsible.Header>
                <Subheader>Pattern</Subheader>
              </Collapsible.Header>
              <Collapsible.Content>
                <img alt="" src="http://subtlepatterns2015.subtlepatterns.netdna-cdn.com/patterns/swirl_pattern.png" />
              </Collapsible.Content>
            </Collapsible>
          </Legend.Panel>
        </Legend>
      </ThemeProvider>
    </div>
  ));
