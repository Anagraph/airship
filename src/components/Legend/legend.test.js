import React from 'react';
import Legend from './legend';
import renderer from 'react-test-renderer';

describe('render', () => {
  it('renders without crashing', () => {
    const component = renderer.create(
      <Legend>
        <Legend.Panel>Hello</Legend.Panel>
      </Legend>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with size', () => {
    const component = renderer.create(
      <Legend small>
        <Legend.Panel>Hello</Legend.Panel>
      </Legend>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
