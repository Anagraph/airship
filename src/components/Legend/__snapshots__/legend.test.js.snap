// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders with size 1`] = `
.c0 {
  border-radius: 4px;
  background-color: #fff;
  width: 160px;
  box-shadow: 0 2px 8px 0 rgba(44,44,44,0.16);
}

.c1 {
  padding: 26px 20px;
  position: relative;
}

.c1:before {
  content: '';
  height: 2px;
  left: 0;
  right: 0;
  top: 0;
  background-color: #F5F5F5;
  position: absolute;
}

.c1:first-child:before {
  display: none;
}

<div
  className="c0"
  style={undefined}
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;

exports[`render renders without crashing 1`] = `
.c0 {
  border-radius: 4px;
  background-color: #fff;
  width: 260px;
  box-shadow: 0 2px 8px 0 rgba(44,44,44,0.16);
}

.c1 {
  padding: 26px 20px;
  position: relative;
}

.c1:before {
  content: '';
  height: 2px;
  left: 0;
  right: 0;
  top: 0;
  background-color: #F5F5F5;
  position: absolute;
}

.c1:first-child:before {
  display: none;
}

<div
  className="c0"
  style={undefined}
>
  <div
    className="c1"
  >
    Hello
  </div>
</div>
`;
